wm/README

```diff
@@ -1 +1,167 @@
-your code here...
+/*
+ * To change this license header, choose License Headers in Project Properties.
+ * To change this template file, choose Tools | Templates
+ * and open the template in the editor.
+ */
+package assignment;
+
+import java.util.ArrayList;
+import java.io.*;
+import java.util.Arrays;
+
+/**
+ *
+ * @author waqar
+ */
+import java.util.Scanner;
+import java.util.StringJoiner;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+public class Assignment {
+
+    /**
+     * @param args the command line arguments
+     */
+    public static void main(String[] args) {
+        try {
+            // TODO code application logic here
+            System.out.println(parse("/home/waqar/Git-Repo/5030/tests/cases.tsv"));
+        } catch (Exception ex) {
+            Logger.getLogger(Assignment.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+ 
+    public static ArrayList<String> read_file(String file_path) throws FileNotFoundException{
+        
+        File fi = new File(file_path);
+        
+        if(!fi.exists())
+            throw new FileNotFoundException();
+        
+        Scanner sc = new Scanner(fi);
+        ArrayList<String> data = new ArrayList<String>();
+        while(sc.hasNextLine()){
+            String data_line = sc.nextLine();
+            data.add(data_line);
+        }
+        
+        return data;
+    }
+    
+    public static ArrayList<String> parse(String file_path) throws FileNotFoundException, IOException, Exception{
+    
+        ArrayList<String> data_list = read_file(file_path);
+        ArrayList<String> filtered_data = new ArrayList<String>();
+        for(String dat: data_list){
+            
+            String[] entry = dat.split("\t");
+            
+            String lang_symbol = entry[0];
+            String[] data = entry[1].split(" ");
```

Comment:
Could cause filtering to fail if there are two spaces next to each other.

Reply:
Space is considered as a single character no matter how many consecutive spaces there are it will work fine.